src/opm/parser/eclipse/EclipseState/Tables/TableManager.cpp

```diff
@@ -1195,6 +1208,114 @@ namespace Opm {
+        };
+
+        const auto formatHeader = [&pvtoLoc](const std::size_t pvtnum)
+        {
+            return fmt::format("Non-Monotonic Oil Formation Volume Factor Detected in Keyword PVTO, PVTNUM={num}\n"
+                               "In {file} line {line}",
+                               "num"_a = pvtnum,
+                               "file"_a = pvtoLoc.filename,
+                               "line"_a = pvtoLoc.lineno);
+        };
+
+        const auto formatBoRecord =
+            [&usys](const std::size_t            numDigitsRecordID,
+                    const std::size_t            floatPrecision,
+                    const PvtoTable::FlippedFVF& flipped)
+        {
+            return fmt::format(
+                "Record {rec:{width}}: "
+                "FVF {BO_1:.{prec}f} at RS {RS_1:.{prec}f} "
+                "is not greater than "
+                "FVF {BO_0:.{prec}f} at RS {RS_0:.{prec}f}",
+                "rec"_a = flipped.i + 1,
+                "width"_a = numDigitsRecordID,
+                "prec"_a = floatPrecision,
+                "BO_1"_a = usys.from_si(M::oil_formation_volume_factor, flipped.Bo[1]),
+                "RS_1"_a = usys.from_si(M::gas_oil_ratio, flipped.Rs[1]),
+                "BO_0"_a = usys.from_si(M::oil_formation_volume_factor, flipped.Bo[0]),
+                "RS_0"_a = usys.from_si(M::gas_oil_ratio, flipped.Rs[0])
+            );
+        };
+
+        std::ostringstream prt;
+        std::ostringstream console;
+
+        {
+            const auto header = formatHeader(tableID);
+            prt     << header << '\n';
+            console << header << '\n';
+        }
+
+        // Append record to console message if either of the following conditions hold
+        //
+        //   1. Total number of flipped records does not exceed `consoleRecordLimit`.
+        //
+        //   2. Record is among `consoleRecordLimit - 1` first records.
+        //
+        // In the second case, also emit limiter message as `consoleRecordLimit`-th
+        // record.
+        //
+        // Print file gets all flipped records.
+        const auto numDigitsRecordID  = nDigit(flipped_Bo.back().i + 1);
+        const auto numRecords         = flipped_Bo.size();
+        const auto consoleRecordLimit = decltype(numRecords){4};
+        const auto floatPrecision     = std::size_t{3};
+        const auto consoleWriteAll    = numRecords <= consoleRecordLimit;
+        for (auto recordIx = 0*numRecords; recordIx < numRecords; ++recordIx) {
+            const auto record =
+                formatBoRecord(numDigitsRecordID, floatPrecision, flipped_Bo[recordIx]);
+
+            prt << record << '\n';
+
+            if (consoleWriteAll || (recordIx + 1 < consoleRecordLimit)) {
+                console << record << '\n';
+            }
+            else if (recordIx + 1 == consoleRecordLimit) {
+                console << "Report limit reached, see PRT-file for additional records.\n";
```

Comment:
Is there a reason for not using the message limiter functionality which is already in OpmLog?

Reply:
I will try to answer this, to see if I understood it correctly: The whole part here just prints a single message to each of the logs, that can have many lines (one per record). The code here limits the lines in the console version of the message. The message limits would only apply if each record was a separate message.

@bska please shoot this down if wrong...

Reply:
The limiter addresses the number of messages, not the number of lines per individual message.

Reply:
> I will try to answer this, to see if I understood it correctly

@atgeirr: Yes, that is correct.